api/app/utils/review.inference.js

```diff
@@ -3,17 +3,38 @@ const db = require("../sequelize/models");
 const Op = db.Sequelize.Op;
 
 const unixEpochTimeInMilliseconds = getUnixTimeFromHours(config.expiryTimeInHours)
-const timeStringAt24HoursPrior = new Date(new Date().getTime() - unixEpochTimeInMilliseconds).toISOString()
-const isActiveCondition = { 
-  mergedAt: { [Op.eq]: null },
-  createdAt: { [Op.gte]: timeStringAt24HoursPrior }
-};
 
-const isInActiveCondition = {
-  [Op.or]: [
-    { createdAt: { [Op.lt]: timeStringAt24HoursPrior } },
-    { mergedAt: { [Op.not]: null } }
-  ]
+const buildIsActiveCondition = (currentTime) => {
+  const timeStringAt24HoursPrior = new Date(currentTime - unixEpochTimeInMilliseconds).toISOString();
+  return { 
+    createdAt: { [Op.gte]: timeStringAt24HoursPrior },
+    mergedAt: { [Op.eq]: null }, // no mergedAt
+    archivedAt: { [Op.eq]: null } // no archivedAt
+  };
+}
+const buildIsPendingCondition = (currentTime) => {
+  const timeStringAt24HoursPrior = new Date(currentTime - unixEpochTimeInMilliseconds).toISOString();
+  return { 
+    createdAt: { [Op.lt]: timeStringAt24HoursPrior }, // expired
+    submittedAt: { [Op.not]: null }, // has been submitted
+    mergedAt: { [Op.eq]: null }, // no mergedAt
+    archivedAt: { [Op.eq]: null } // no archivedAt
+  };
+}
+
+const buildIsInActiveCondition = (currentTime) => {
+  const timeStringAt24HoursPrior = new Date(currentTime - unixEpochTimeInMilliseconds).toISOString();
+  return { 
+    [Op.or]: {
+      mergedAt: { [Op.not]: null }, // has been merged
+      archivedAt: { [Op.not]: null }, // has been archived
+      // inactive conditions when review has expired
+      [Op.and]: {
```

Comment:
if we add this condition, how do we ascertain that an inactive transcript should be re-queued (given that it has expired but not submitted)? Is this accounted for somewhere else in the code base?

Reply:
These conditions are for getters. e.g. Get all inactive from a `/reviews` API call. i.e find all reviews that this condition satisfies

Requeue is done from either of 2 processes:
- CRON job: A cron is created once a transcript is claimed (review is created) we pass the reviewId to the CRON and it is scheduled to run exactly 24 hours after it has been created. It doesn't use this condition as we don't do a findAll, rather get that review from the reviewId then if it has any timeStamp (submittedAt, mergedAt, archivedAt) don't requeue the transcript associated with that review. 
- Webhook: If the PR opened against a review is closed by the repo maintainer (not merged) find the review associated with that PR_URL and requeue the associated transcript